File: android/src/main/java/com/trentrand/plugins/instant/InstantPlugin.java
package com.trentrand.plugins.instant;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "Instant")
public class InstantPlugin extends Plugin {
    private Instant implementation;

    @Override
    public void load() {
        implementation = new Instant(getContext());
    }

    @PluginMethod
    public void isInstantApp(PluginCall call) {
        JSObject ret = new JSObject();
        ret.put("value", implementation.isInstantApp());
        call.resolve(ret);
    }
}

File: ios/Sources/InstantPlugin/InstantPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(InstantPlugin)
public class InstantPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "InstantPlugin"
    public let jsName = "Instant"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "isInstantApp", returnType: CAPPluginReturnPromise)
    ]
    private let implementation = Instant()

    @objc func isInstantApp(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isInstantApp()
        ])
    }
}
